package.json

```diff
@@ -13,8 +13,8 @@
     "gen": "gocog -c=node -a=%s -a=-p ./src/**.ts && ts-standard --fix",
     "build": "tsc",
     "build:watch": "tsc --watch",
-    "test": "$npm_execpath run build && jest",
-    "test:watch": "$npm_execpath run build:watch & jest --watchAll"
+    "test": "npm run build && jest",
+    "test:watch": "npm run build:watch & jest --watchAll"
```

Comment:
I was trying to make this reference the "default" npm for user's shell environment, but I gues npm itself is always there. Should we add https://www.npmjs.com/package/only-allow and add other lockfile names to .gitignore?

Reply:
To focing users to use specific pm, how about using corepack? Is it different from `allow-only` library?

Reply:
I haven't used corepack before, seems like it makes npm "use" the preferred package manager specified in package.json in the actual installation process but it does not update the corresponding lockfile. Maybe I'm doing something wrong. Meanwhile, allow-only remains to be a hassle-free solution for enforcing a single package manager.

to setup, you have to add this line to scripts object in package.json
```json
{
	...
	"scripts": {
        "preinstall": "npx only-allow npm",
		...
	},
	...
}
```

It prevents using other package managers by returning with error exit code when different package manager is used to invoke install command

Reply:
If `package.json` has the following field
```json
{
    "packageManager": "npm@7.11.2"
}
```

Following error occurs when user tried to use any yarn command
```pwsh
PS C:\dev\josa-complete> yarn
Usage Error: This project is configured to use npm

$ yarn ...
```

What is lacking when using "Corepack" compared to "allow-only"?

Reply:
The main difference is that `only-allow` additionally supports `cnpm` and gives installation instructions when giving an error. It is a matter of preference.
If we are going to use corepack, should the `"preinstall": "corepack enable"` suffice, then?

On my end, invoking yarn just works even though `"packageManager": "npm@7.11.2"` is set and `corepack enable` is run just before it.

Reply:
Oh I didn't know that corepack is an experimantal feature. Using `allow-only` would be proper right now👍